client/services/code_preprocessors/CodePreprocessorDispatcherServiceSpec.js

```
@@ -0,0 +1,55 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the CodePreprocessorDispatcherService.
+ */
+
+describe('CodePreprocessorDispatcherService', function() {
+  var CodePreprocessorDispatcherService;
+  var PythonCodePreprocessorService;
+  var LANGUAGE_PYTHON;
+  var CodeSubmissionObjectFactory;
+
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    CodePreprocessorDispatcherService = $injector.get(
+      'CodePreprocessorDispatcherService');
+    PythonCodePreprocessorService = $injector.get(
+      'PythonCodePreprocessorService');
+    LANGUAGE_PYTHON = $injector.get('LANGUAGE_PYTHON');
+    CodeSubmissionObjectFactory = $injector.get('CodeSubmissionObjectFactory');
+  }));
+
+  describe('preprocess', function() {
+    it('should throw error of Language not supported', function() {
```

should throw an error if the language passed in is not supported